src/btree/bt_page.c

```diff
@@ -1311,8 +1317,12 @@ __wt_page_alloc(WT_SESSION_IMPL *session, uint8_t type, uint32_t alloc_entries,
 
     /* Increment the cache statistics. */
     __wt_cache_page_inmem_incr(session, page, size, false);
-    (void)__wt_atomic_add_uint64(&S2C(session)->cache->pages_inmem, 1);
-    page->cache_create_gen = __wt_atomic_load_uint64_relaxed(&S2C(session)->evict->evict_pass_gen);
+    (void)__wt_atomic_add_uint64(&cache->pages_inmem, 1);
+    if (F_ISSET(btree, WT_BTREE_GARBAGE_COLLECT))
```

Comment:
How about changing this into nested if to limit the code checks for the ASC?
if (disagg) {
   if (garbage collect)
    ...
    else
    ....
}

Reply:
ok, I'll change that.

Reply:
done